src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -209,48 +211,201 @@ internal static IEnumerable<string> GetCommaSeparatedValues(string text)
             var accessibleProperties = type.GetRuntimeProperties().Where(IsAccessibleProperty);
             var attributeDescriptors = new List<TagHelperAttributeDescriptor>();
 
+            // Keep indexer descriptors separate to avoid sorting the combined list later.
```

Comment:
I thought we have it planned to explode in the case that you utilize both mechanisms of setting dictionary attributes.